src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/Merge/CompositeSearchParamRowGenerator.cs

```diff
@@ -0,0 +1,83 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Microsoft.Health.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using LinqExpression = System.Linq.Expressions.Expression;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage.TvpRowGeneration
+{
+    internal abstract class CompositeSearchParamRowGenerator<TSearchValue, TRow> : MergeSearchParameterRowGenerator<TSearchValue, TRow>
+        where TSearchValue : ITuple
+        where TRow : struct
+    {
+        private readonly Func<EnumeratorWrapper<ISearchValue>, TSearchValue> _converter = CreateConverterFunc();
+
+        protected CompositeSearchParamRowGenerator(SqlServerFhirModel model, SearchParameterToSearchValueTypeMap searchParameterTypeMap)
+            : base(model, searchParameterTypeMap)
+        {
+        }
+
+        protected override IEnumerable<TSearchValue> ConvertSearchValue(SearchIndexEntry entry)
+        {
+            var compositeSearchValue = (CompositeSearchValue)entry.Value;
```

Comment:
Should we check if entry is null?

Reply:
This is the behavior of the previous code.